tfjs-converter/src/executor/model_analysis.ts

```diff
@@ -146,26 +145,87 @@ export function getNodesInTopologicalOrder(
   return orderedNodes;
 }
 
-const CONTROL_FLOW_OPS = [
+/**
+ * Given the execution info, return a map from node to the disposable node list
+ * after its execution.
+ *
+ * @returns A map from node to disposable nodes after its
+ *     execution. That is, for a node `x`, `nodeLiveUntilMap[x]` indicates all
+ *     nodes which their intermediate tensors should be disposed after `x` being
+ *     executed.
+ */
+export function getNodeLiveUntilMap(orderedNodes: Node[]): Map<Node, Node[]> {
+  const nNodes = orderedNodes.length;
+  const nodeToOrder = new Map<Node, number>();
+  for (let i = 0; i < nNodes; ++i) {
+    nodeToOrder.set(orderedNodes[i], i);
+  }
+
+  // `liveUntil[i]` indicates that "all the intermediate tensors from
+  // `orderedNodes[i]` should be disposed after `orderedNodes[liveUntil[i]]`
+  // being executed."
+  const INF_LIFE = Math.floor(Number.MAX_SAFE_INTEGER / 2);
```

Comment:
Nit: Pull this out of the function maybe? It probably doesn't matter.

Reply:
Is there any other functions use the live until info? I'm thinking to rewrite the old `checkNodesForDisposal` with this info, but I don't have any good ideas so far. Maybe later.

Reply:
I was talking about the const `INF_LIFE`, but github highlighted a bunch of other lines too. It can probably stay in the function, though.

Reply:
Then I'd just leave it here. This analysis does not expose the lifespan in numbers, so it's better to keep this intermediate value inside.